activity/src/main/java/com/amazonaws/stepfunctions/cloudformation/activity/ResourceHandler.java

```diff
@@ -61,4 +61,13 @@
         return resultBuilder.build();
     }
 
+    protected void validateResourceArn(String resourceArn) {
+        if (resourceArn == null) {
+            AmazonServiceException exception = new AmazonServiceException(Constants.INVALID_ARN_ERROR_CODE);
+            exception.setStatusCode(400);
+            exception.setErrorCode(Constants.INVALID_ARN_ERROR_CODE);
+            throw exception;
+        }
+    }
```

Comment:
+1 to the purpose matching wording. A full validation on the arn might be hard, but we can at least change the null check exception (may add empty string check) wording as "ResourceArnNotFound" ("not exist"/" not provided" etc if we wanna provide detailed msg ) for better visibility. 
Not for this CR but we might can use https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/arn/Arn.html later to build ARNs with "fromString" method so that we only need to do a null check.

Reply:
Renamed method to `verifyActivityArnIsPresent`. Added a new error message to communicate the ARN is null. I think it's okay to let the API validate the length of the ARN string as long it's not null.